### DISCUSSION TITLE
WebGL Blob

### DISCUSSION THREAD

**Discussion by MonsieurD** [videokit-ai/natcorder#20] [Q&A] [answered]
Hello !

I'm recording a video from my game camera. I use the webm recorder on webgl platform. When the record is finished I get a blob url. 

I download the blob with UnityWebRequest :
```
  UnityWebRequest webRequest = UnityWebRequest.Get(url);

  yield return webRequest.SendWebRequest();

  if (webRequest.result != UnityWebRequest.Result.Success)
  {
      Debug.LogError(webRequest.error);
  }
  else
  {
      Debug.Log("Web request ok");

      SaveToFile(webRequest.downloadHandler.data);
  }
```
Then I save file to persistent data path :
```
private void SaveToFile(byte[] data)
{
	var today = System.DateTime.Now;

	string fileName = $"Record_{today.Year.ToString("00")}_{today.Month.ToString("00")}_{today.Day.ToString("00")}_{today.Hour.ToString("00")}_{today.Minute.ToString("00")}_{today.Second.ToString("00")}.webm";

	string filePath = Path.Combine(Application.persistentDataPath, fileName);

	File.WriteAllBytes(filePath, data);

	Debug.Log($"Data saved to {filePath}");

	Application.OpenURL(filePath);
}
```
Now I have a file into this folder : /idbfs/8b491ca5a5c6d71381c5be0084e4333d/Record_2023_01_17_10_45_01.webm
But how can I make the user to download this file ?
The application.openURL open an AccessDenied page.

Someone can help me ?

**Comment by olokobayusuf** [comment]
Hey @MonsieurD are you trying to download the WEBM file to the user's computer? If so, you'll need to write a `jslib` plugin to do this. We can look into adding this feature into NatCorder in the next release.

**Reply by MonsieurD** [reply]
Yes I try to download the webm file to the user's computer. I'm very bad at writting js code so if you have ever done this thing before and can share the code I will be grateful.

**Comment by olokobayusuf** [comment, marked as answer]
Hey @MonsieurD I totally forgot about this, but you should actually be able to use [NatShare](https://github.com/natmlx/NatShare) to download the blob:
```csharp
// Finish writing
var recordingPath = await recorder.FinishWriting();
// Use NatShare to download the video
var payload = new SavePayload();
payload.AddMedia(recordingPath);
await payload.Save();
```
Try this and let me know if you face any issues.

**Reply by olokobayusuf** [reply]
I totally forgot about this, but NatShare's `SavePayload` supports downloading blobs from the browser to the user's computer.
